stdlib/SuiteSparse/src/umfpack.jl

```diff
@@ -406,6 +406,32 @@ for itype in UmfpackIndexTypes
                         mx, mz, C_NULL, lu.numeric, umf_info)
             complex(mx[], mz[])
         end
+        function logabsdet(F::UmfpackLU{T, $itype}) where T # return log(abs(det)) and sign(det)
+            n = checksquare(F)
+            issuccess(F) || return log(zero(real(T))), log(one(T))
+            U = diag(F.U)
+            Rs = F.Rs
+            p = F.p
+            q = F.q
+            c = 0
+            P = one(T)
+            abs_det = zero(real(T))
+            @inbounds for i in 1:n
+                dg_ii = U[i] / Rs[i]
+                P *= sign(dg_ii)
+                for j in i+1:n
+                    if p[i] > p[j]
+                        c += 1
```

Comment:
I'm not sure it's relevant, but one could avoid the risk of integer overflow by initializing `c = 1` above and then have here `c = flipsign(c, -1)`, and finally `s = c * one(real(T)) * P` in the last line? The difference to the generic method is that we sum over `p` and `q` and in two loops ("O(n^2)"), which could potentially be way more than just a single pass through `ipiv` as in the generic method.

Reply:
Just to make sure I understand, your concern is that if the matrix is too big, `c` might overflow as it's incremented inside the nested loops over `p` and `q`?

Reply:
Exactly! But the `flipsign`-stuff is too complicated, not sure why I came with this in the first place. Just `c = -c` would suffice, of course.

Reply:
Well, or even have a `Bool` switch, starting from `c = false`, then update `c != c` and finally have the `ifelse(c, -one(real(T)), one(real(T))) * P`. Something like that.

Reply:
Ok yeah, that sounds reasonable.